core/src/main/scala/org/allenai/common/SeekableSource.scala

```diff
@@ -0,0 +1,256 @@
+package org.allenai.common
+
+import java.nio.ByteBuffer
+import java.nio.channels.FileChannel
+import java.util.NoSuchElementException
+
+import scala.io.Codec
+
+/** A class that provides seekability to an interface similar to scala.io.Source. This is required
+  * because scala.io.Source only provides character offsets, while FileChannel (and other seekable
+  * Java interfaces) take byte offsets for seeking. The only limitation is that the file must be
+  * either UTF-8 or ISO-8859-1 encoded.
+  * <p>This also has a `getLines()` method to iterate over lines. Any reads done on this iterator
+  * are reflected in the main Source, and reads on the Source are reflected in the
+  * `Iterator[String]`.</p>
+  * @param inFile the file channel to wrap
+  * @param bufferSize the size of the internal buffer to use
+  * @param codec the codec to use. Must be one of UTF-8 or ISO-8859-1.
+  * @throws IllegalArgumentException if `bufferSize` is less than 3, or `codec` is not UTF-8 or
+  * ISO-8859-1
+  */
+class SeekableSource(inFile: FileChannel, bufferSize: Int = 8192)(implicit codec: Codec)
```

Comment:
I think I'm just living in the past . . . :P

This size is what `java.io.BufferedReader` uses. Otherwise, it's pretty arbitrary. I wasn't seeing huge boosts when I was playing around with sizes, but 8M seems like a totally sensible default.